### ISSUE TITLE
Share inaccessible (Connect as fails), but daemon works, etc.

### ISSUE THREAD

**Issue by bugsyb** [mbentley/docker-timemachine#105]
Not sure if problem is not related to a move to arm64v8/alpine:latest as base image. It removed the other problem mentioned  here: https://github.com/mbentley/docker-timemachine/issues/81

Every time trying to mount the share (Connect as) it fails and talks about server being inaccessible. Tcpdump shows proper connection and it is at application layer.
Logs showed below for each connection:
```
tdb(/var/cache/samba/gencache.tdb): tdb_mutex_open_ok[/var/cache/samba/gencache.tdb]: Mutex size changed from 241664 to 401408
tdb(/var/cache/samba/gencache.tdb): tdb_mutex_open_ok[/var/cache/samba/gencache.tdb]: Mutex size changed from 241664 to 401408
tdb(/var/cache/samba/gencache.tdb): tdb_mutex_open_ok[/var/cache/samba/gencache.tdb]: Mutex size changed from 241664 to 401408
```

Removing gencache.tdb seems to fixed the problem, at least can connect as to the share.

After another container restart below has popped up:
```
Failed to fetch record!
pcap cache not loaded
error in mds_init_ctx for: /opt/user
_mdssvc_open: Couldn't create policy handle for TimeCapsule-user
```
It is still possible to connect as though opening mounted folders (Backups.backupdb) takes ages and produces more of
```
Failed to fetch record!
pcap cache not loaded
```

This is probably due to deletion of gencache.tdb, so it rebuilds it, hence slowness, can see very low I/O activity at the disk (400-1500kB/s) and on outgoing interface related volume of data being pumped out back to Mac (below is iostat (30)
```
Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
sda              14.00      1648.00         0.00       1648          0
```
This suggest more or leas that it might be a lot of hdd head movement, though iowait is not that high either. So a lot of small reads I take it.

There is already ~400GB of backup files, so not sure if it's not going through all of that and not sure how to speed it up.
Glanced at: https://github.com/mbentley/docker-timemachine/issues/75, which can be related.

Any hints on what to look at or to speed it up or potential sources and ideas how it could be sped up if at all?

**Comment by mbentley** [comment]
I wish I could say that I had more information for you but I only know enough about Samba to create this image and for the most part, I am just following the documented configurations suggested for using Samba as a time machine target.

**Comment by cc13com** [comment]
Hi, 

I run in the same/similar issue. After one of the latest updates (don't know if it was my QNAP firmware or the Time Machine docker on this NAS), my Mac will not longer connect with the Time Machine.

If I try to connect via Mac OS system settings I see the following message in the docker-log:

```
Registering new address record for 192.168.1.5 on eth0.IPv4.
Registering new address record for 127.0.0.1 on lo.IPv4.
Failed to fetch record!
Server startup complete. Host name is cdd860b2446e.local. Local service cookie is 1489382787.
Service "cdd860b2446e" (/etc/avahi/services/smbd.service) successfully established.
*****
Samba name server CDD860B2446E is now a local master browser for workgroup WORKGROUP on subnet 192.168.1.5
*****
error in mds_init_ctx for: /opt/timemachine
_mdssvc_open: Couldn't create policy handle for TimeMachine
```

Any idea what I can do to solve this?

Thanks,
CC13

**Comment by mbentley** [comment]
Those logs look normal.  How are you launching the container and what messages are you getting from your Mac?

**Comment by mbentley** [comment]
Heh no worries about the German 😄 Thanks for the compose that you're using along with the translation of the error.  That would have me narrow it down to the `SMB_VFS_OBJECTS` variable.  The default value is `acl_xattr fruit streams_xattr` where the current value you're setting is `fruit streams_xattr`.  There was discussion in https://github.com/mbentley/docker-timemachine/issues/69 about the addition of the `acl_xattr` attribute.  I might suggest updating the `SMB_VFS_OBJECTS` to the default and giving it another try to run.

If that still doesn't work, have you updated your macOS recently or anything that might have changed since the last backup was successful?  One thing you can try is to use an older image.  Find the last date that your backup was successful and update the image tag to match.  So if the last backup was successful on January 2nd, 2022, try the `smb-20220102` tag and see if that succeeds.

**Comment by cc13com** [comment]
I updated to Monterey 12.2 in the last days, after my last time machine backup.

Changed the `SMB_VFS_OBJECTS` to the default as you mentioned. Same issue.

Changed the image to ` image: mbentley/timemachine:smb-20220117` because the January 17th, 2022 was the last day when the backup run fine. Same issue. 

I also tried right now your time-machine-docker-compose template for afp. But it will also not work. I guess there is a problem with my QNAP (where Docker is running). I also updated this in the last few weeks. Will check there.

**Comment by mdbraber** [comment]
@cc13com @mbentley has this issue been resolved? I'm running into this as well. An initial backup works, but after a few days it starts to fail with these messages as well:

```
error in mds_init_ctx for: /opt/timemachine
_mdssvc_open: Couldn't create policy handle for TimeMachine-mdbraber
```

**Comment by cc13com** [comment]
@mdbraber no and I didn't have the time for more investigation.

**Comment by mbentley** [comment]
> error in mds_init_ctx for: /opt/timemachine
> _mdssvc_open: Couldn't create policy handle for TimeMachine-mdbraber


These aren't critical errors for time machine on Samba.  I haven't looked into it further as I don't have similar hardware - I only use the image for amd64.  Just to check, are you using any non-default configuration parameters than what is [defined in the README](https://github.com/mbentley/docker-timemachine#example-usage-for-smb) outside of like username/password? 

It also may be possible to see critical error messages by setting `DEBUG_LEVEL` to `3` or maybe `5` - I don't recall which level provides more info.  I know in the past, there have been some arm related issues in the upstream packages, at least on armv7l but I'm not a Samba developer (or really any sort of developer to be honest) so I have relatively limited visibility into Samba itself.

**Comment by mdbraber** [comment]
> These aren't critical errors for time machine on Samba. I haven't looked into it further as I don't have similar hardware - I only use the image for amd64. Just to check, are you using any non-default configuration parameters than what is [defined in the README](https://github.com/mbentley/docker-timemachine#example-usage-for-smb) outside of like username/password?

No - not using any other parameters.

> It also may be possible to see critical error messages by setting DEBUG_LEVEL to 3 or maybe 5 - I don't recall which level provides more info. I know in the past, there have been some arm related issues in the upstream packages, at least on armv7l but I'm not a Samba developer (or really any sort of developer to be honest) so I have relatively limited visibility into Samba itself.

I'll see if I can figure out more by increasing the debug level.

**Comment by hollie** [comment]
Hey all,

I was also running in to errors to connect to the shares from Finder after updating to Monterey.

Adding this line to the smb.conf global settings and rebooting the Mac resolved the issue for me:

```
server min protocol = SMB2
```

Best regards,
 Lieven

**Comment by mbentley** [comment]
Hmm, so where did you add that? On the mac?  In the time machine container itself?  Just checking since it is already set in the time machine container's [default `smb.conf` that gets generated as you can see here](https://github.com/mbentley/docker-timemachine/blob/master/entrypoint.sh#L226).

**Comment by hollie** [comment]
Hey Matt,

sorry if it was not clear: I added it to the `smb.conf` custom configuration that I have in the container settings.

And I agree that the content is in the autogenerated file. Apparently is was not in the file where I got my initial configuration from, so the problem was local to my setup. Sorry for the confusion, I just thought I could bring a suggestion that would help...

Best regards,
 Lieven
